src/entry.ts

```diff
@@ -198,24 +230,77 @@ class Entry {
     }
     // Format httpRequest
     const req = this.metadata.httpRequest;
-    if (
-      req &&
-      ('statusCode' in req ||
-        'headers' in req ||
-        'method' in req ||
-        'url' in req)
-    ) {
-      entry.httpRequest = makeHttpRequestData(req);
+    if (isRawHTTP(req)) {
```

Comment:
nit: this function is getting a bit long. It might be worth breaking some of this out into helpers, especially if more cases are coming later

Reply:
it IS long. I'll refactor out the timestamp conversion to util/common. Would want to keep the httprequest parsing stuff in inline so people dont miss out on whats happening here.